### DISCUSSION TITLE
Upgrade SlateJS

### DISCUSSION THREAD

**Discussion by dimaip** [keystonejs/keystone#5316] [Feature requests]
Currently Keystone uses SlateJS v0.47, which is not compatible with modern Slate plugins and data format.
There's a migration script for migrating data.

**Comment by JedWatson** [comment]
Hi @dimaip,

We've looked into upgrading the version of Slate used in the `Content` field before and it's unfortunately nontrivial, so I'm afraid that specific field is not going to get updated at this point.

We have, however, been working on a **New** `Document` field which is the replacement for the content field in the next version of Keystone, and it's built on the current version of Slate with some really great new features.

You can see a preview here: https://next.keystonejs.com/guides/document-fields

**Reply by gautamsi** [reply]
would there be migration path?
